### DISCUSSION TITLE
Rebase button online

### DISCUSSION THREAD

**Discussion by RDIL** [community/community#22281] [Pull Requests]
<p>I know there is a branch protection rule for this, but I think there should be a button for people to rebase their forks  online with ease.  Any thoughts?  </p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/rdil">@rdil</a>,</p>
<p>Thanks for this feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Please let me know if you have any other questions.</p>
<p>Cheers!</p>

**Comment by majormoses** [comment]
<p>There is now an option to merge master into the branch, while a merge is not quite the same as a rebase and I personally do prefer rebasing its important to keep in mind that rebasing should not be done on a shared branch with other developers using it.</p>

**Comment by gh-jvirdee** [comment]
Is rebase available now?

**Reply by david-bascom-ihm** [reply]
Yes.
1. Open your PR
2. Click the ‘Edit’ button to the right of your PR title.
3. The base branch now appears as a dropdown. 
4. Select desired branch and continue.

**Reply by anr2me** [reply]
> Yes.
> 
>     1. Open your PR
> 
>     2. Click the ‘Edit’ button to the right of your PR title.
> 
>     3. The base branch now appears as a dropdown.
> 
>     4. Select desired branch and continue.

This is an incomplete guide, the only way to rebase a PR is (AFAIK):
1. Create a new branch based on the updated base branch (ie. master/main)
2. Open your PR
3. Click the ‘Edit’ button to the right of your PR title.
4. The base branch now appears as a dropdown.
5. Select the new branch from step-1 as the new base branch and continue.
6. Select ‘Rebase and merge’ at the bottom of the PR.
7. Create a new PR based on the new branch from step-1.

Then continue working on the new (drafted) PR.

PS: Quite a hassle doing it directly from github website, i wished we can Rebase a PR without merging it first, thus doesn't need to recreated the PR.

For example, if only by temporarily/simply switching the base branch from master/main branch to other branch, and then goes back to the (updated) master/main branch can automatically rebase the PR with the updated base branch.
